python/shotgun_fields/image_widget.py

```diff
@@ -279,24 +318,53 @@ def _replace_image(self):
         """
         self._upload_image()
 
-    def _scale_pixmap(self):
+    def _scale_pixmap(self, pixmap):
         """
         Scale the pixmap in preparation for display.
         """
-        scaled_pixmap = self._pixmap.scaled(
+        return pixmap.scaled(
             self.size(),
             QtCore.Qt.KeepAspectRatio,
             QtCore.Qt.SmoothTransformation
         )
-        self._scaled_width = scaled_pixmap.width()
-        super(ImageWidget, self).setPixmap(scaled_pixmap)
-        self._update_btn_position()
 
     def _show_image(self):
         """
-        The user requested to show the file. Forward to desktop services.
+        The user requested to show the file.
+
+        Display the image in a transient, modeless QDialog.
         """
-        QtGui.QDesktopServices.openUrl("file:///%s" % (self._image_url))
+
+        # don't continue unless there's somethign to show
+        if not self._pixmap:
+            return
+
+        if self._image_url:
```

Comment:
Yes, thanks! I have deprecated the `image_url` property and added `image_path` instead. I got rid of the previous call to `QDesktopServices` because of the expiry issue. Now it pops up a Qt dialog displaying the downloaded image path if there is one. Hope that makes sense!